third_party/amd/lib/TritonAMDGPUToLLVM/ElementwiseOpToLLVM.cpp

```diff
@@ -433,35 +433,71 @@ static Value convertBf16ToFp32(Location loc,
   return bitcast(shifted, f32_ty);
 }
 
+static Value buildGCNInstruction(Location loc,
+                                 ConversionPatternRewriter &rewritter,
+                                 const std::string &instr_name,
+                                 const SmallVector<std::string> &constraints,
+                                 const SmallVector<Value> &vals,
+                                 Type ret_type) {
+  assert(constraints.size() == vals.size() + 1);
+  assert(vals.size() == 2 or vals.size() == 3);
+  GCNBuilder builder;
+  auto &instr = *builder.create(instr_name);
+  auto out = builder.newOperand(constraints[0]);
```

Comment:
Specify the result type if it's not too complicated. `auto` is bad for readability. 

Similarly for other `auto` in the next function.